src/consumer/consumerGroup.js

```diff
@@ -350,7 +352,7 @@ module.exports = class ConsumerGroup {
         })
 
         await sleep(this.maxWaitTime)
-        return []
+        return EMPTY_FETCH_RESULT
```

Comment:
It's better to return a new instance here, iterators have an internal state.

Reply:
Given that we want a "done" iterator for this case, isn't that a benefit?